pkg/batchd/policy/drf/drf.go

```diff
@@ -98,6 +102,32 @@ func (drf *drfScheduler) Allocate(queues []*cache.QueueInfo, nodes []*cache.Node
 				break
 			}
 		}
+
+		// If policy could not assign minAvailable for a PodSet, then recycle assigned resources in
+		// this loop even if the PodSet was assigned resources before it is to handle a case that PodSet
+		// meet minAvailable in the past and then become under minAvailable due to some pods finish
+		if !psi.meetMinAvailable() {
+			for podName, nodeName := range assignedPods {
+				pi := psi.resetAndGetAssignedPod(podName)
```

Comment:
prefer to use "flag or mark", "commit if meet, otherwise discard" steps; instead of "add", "sub" :).

Reply:
And we may add some new func/method to PodSetInfo.